### PR TITLE
Add forward AD formulas for mv, scatter_add, _s_where

### PR DESCRIPTION
**Pull request by soulitzer**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #71026
* __->__ #70468
* #70355

Differential Revision: [D33405364](https://our.internmc.facebook.com/intern/diff/D33405364)

### CODE REVIEW

File: tools/autograd/derivatives.yaml [hunk 1]
```diff
@@ -1634,6 +1636,7 @@
   condition: non_differentiable
   self: where(condition, grad, zeros_like(grad))
   other: where(condition, zeros_like(grad), grad)
+  result: where(condition, self_t, zeros_like(other_p)) + where(condition, zeros_like(self_p), other_t)
```
Comment (albanD) [inline tools/autograd/derivatives.yaml:1639]: Why not `where(condition, self_t, other_t)`?

Comment (soulitzer) [reply tools/autograd/derivatives.yaml:1639]: !

**Comment by soulitzer** [comment]
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33405364).

**Review by albanD — APPROVED** [review]
SGTM

File: tools/autograd/derivatives.yaml [hunk 2]
```diff
@@ -1072,6 +1072,7 @@
 - name: mv(Tensor self, Tensor vec) -> Tensor
   self: grad.ger(vec.conj())
   vec: self.conj().t().mv(grad)
+  result: mv(self_t, vec_p) + mv(self_p, vec_t)
```
Comment (anjali411) [inline tools/autograd/derivatives.yaml:1075]: it's not necessary, but I think it would be cool to add a ZeroTensor fastpath for `mv` as a follow-up.

Comment (albanD) [reply tools/autograd/derivatives.yaml:1075]: Yes!

**Comment by soulitzer** [comment]
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33405364).

**Comment by soulitzer** [comment]
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33405364).

**Comment by soulitzer** [comment]
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33405364).

**Comment by soulitzer** [comment]
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33405364).

**Review by albanD — APPROVED** [review]

**Comment by soulitzer** [comment]
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33405364).

**Comment by soulitzer** [comment]
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33405364).

**Comment by Lezcano** [comment]
Now that I see this, I had a PR that's completely stalled due to a mobile error that implemented a number of forward formulas, finally enabling forward AD for matmul and a few other functions. See https://github.com/pytorch/pytorch/pull/64387

**Pull request closed without merging**